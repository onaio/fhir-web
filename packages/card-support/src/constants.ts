// OPENSRP API endpoints
export const OPENSRP_URL_USER_ASSIGNMENT = '/organization/user-assignment';
export const OPENSRP_URL_CLIENT_SEARCH = '/client/search';
export const OPENSRP_URL_LOCATION_HIERARCHY = '/location/hierarchy';

/** CSV File MIME Types */
export const TEXT_CSV = 'text/csv';
export const APPLICATION_CSV = 'application/csv';
export const TEXT_PLAIN = 'text/plain';
export const TEXT_X_CSV = 'text/x-csv';
export const APPLICATION_VND_EXCEL = 'application/vnd.ms-excel';
export const APPLICATION_X_CSV = 'application/x-csv';
export const TEXT_COMMA_SEPARATED_VALUES = 'text/comma-separated-values';
export const TEXT_X_COMMA_SEPARATED_VALUES = 'text/x-comma-separated-values';
export const TEXT_TAB_SEPARATED_VALUES = 'text/tab-separated-values';
