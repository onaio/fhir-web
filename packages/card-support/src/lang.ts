import i18n, { namespace } from './mls';
import { Dictionary } from '@onaio/utils';

export type Lang = Dictionary<string>;
const lang: Lang = {};

/** recompute values */
function fill() {
  lang.ERROR_OCCURRED = i18n.t(`An error occurred`, { ns: namespace });
  lang.DOWNLOAD_CLIENT_DATA = i18n.t(`Download Client Data`, { ns: namespace });
  lang.CLIENT_LOCATION = i18n.t(`Client Location`, { ns: namespace });
  lang.ALL_LOCATIONS = i18n.t(`All Locations`, { ns: namespace });
  lang.CARD_STATUS = i18n.t(`Card Status`, { ns: namespace });
  lang.NEEDS_CARD = i18n.t(`Needs Card`, { ns: namespace });
  lang.CARD_NOT_NEEDED = i18n.t(`Card not needed`, { ns: namespace });
  lang.BOTH_NEEDS_CARD_CARD_NOT_NEEDED = i18n.t(`Both "Needs card" and "Card not needed"`, {
    ns: namespace,
  });
  lang.CARD_ORDER_DATE = i18n.t(`Card Order Date`, { ns: namespace });
  lang.CARD_ORDER_DATE_REQUIRED = i18n.t(`Please enter start date and end date`, { ns: namespace });
  lang.SELECT_CARD_ORDER_DATE = i18n.t(`Select Card Order Date`, { ns: namespace });
  lang.DOWNLOADING = i18n.t(`Downloading`, { ns: namespace });
  lang.DOWNLOAD_CSV = i18n.t(`Download CSV`, { ns: namespace });
  lang.NO_DATA_FOUND = i18n.t(`No data found`, { ns: namespace });
}

// run it initial
fill();

// bind some events and fill values again (doing the magic you expect to happen magically)
i18n.on(`languageChanged`, () => {
  fill();
});

// export the const
export default lang;
