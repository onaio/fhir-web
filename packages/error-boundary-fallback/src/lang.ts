import i18n, { namespace } from './mls';
import { Dictionary } from '@onaio/utils';

const lang: Dictionary<string> = {};

/** recompute values */
function fill() {
  lang.PAGE_TITLE = i18n.t(`An Error Occurred`, { ns: namespace });
  lang.PAGE_SUB_TITLE = i18n.t(
    `There has been an error. Itâ€™s been reported to the site administrators via email and should be fixed shortly. Thanks for your patience.'`,
    { ns: namespace }
  );
  lang.BUTTON_TITLE = i18n.t(`Back Home`, { ns: namespace });
}

// run it initial
fill();

// bind some events and fill values again (doing the magic you expect to happen magically)
i18n.on(`languageChanged`, () => {
  fill();
});

// export the const
export default lang;
