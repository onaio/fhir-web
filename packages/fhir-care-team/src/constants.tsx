// url

// Query patams
export const SEARCH_QUERY_PARAM = 'searchQuery';

// FHIR API strings
export const URL_ADMIN = '/admin';
export const FHIR_CARE_TEAM = 'CareTeam';
export const FHIR_GROUPS = 'Group';
export const FHIR_PRACTITIONERS = 'Practitioner';
export const URL_EDIT_CARE_TEAM = `${URL_ADMIN}/CareTeam/edit`;
export const URL_CREATE_CARE_TEAM = `${URL_ADMIN}/CareTeam/new`;
export const URL_CARE_TEAM = `${URL_ADMIN}/CareTeams`;

// Route params
export const ROUTE_PARAM_CARE_TEAM_ID = 'careTeamId';
