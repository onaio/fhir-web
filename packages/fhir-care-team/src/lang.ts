import i18n, { namespace } from './mls';
import { Dictionary } from '@onaio/utils';

export type Lang = Dictionary<string>;

const lang: Lang = {};

/** recompute values */
function fill() {
  // Tooltips
  lang.TIP_REQUIRED_FIELD = i18n.t(`This is a required field`, { ns: namespace });

  // Errors
  lang.ERROR_OCCURRED = i18n.t(`An error occurred`, { ns: namespace });

  // Messages
  lang.CARE_TEAMS_UPDATE_SUCCESS = i18n.t(`Successfully Updated Care Teams`, { ns: namespace });
  lang.CARE_TEAMS_ADD_SUCCESS = i18n.t(`Successfully Added Care Teams`, { ns: namespace });
  lang.CARE_TEAM_DELETE_SUCCESS = i18n.t(`Successfully Deleted Care Team`, { ns: namespace });
  lang.CONFIRM_DELETE = i18n.t(`Are you sure you want to delete this Care Team?`, {
    ns: namespace,
  });
  lang.YES = i18n.t(`Yes`, { ns: namespace });
  lang.NO = i18n.t(`No`, { ns: namespace });

  // Rendered text
  lang.STATUS = i18n.t(`Status`, { ns: namespace });
  lang.NAME = i18n.t(`Name`, { ns: namespace });
  lang.SAVING = i18n.t(`Saving`, { ns: namespace });
  lang.SAVE = i18n.t(`Save`, { ns: namespace });
  lang.ACTIVE = i18n.t(`Active`, { ns: namespace });
  lang.INACTIVE = i18n.t(`Inactive`, { ns: namespace });
  lang.CANCEL = i18n.t(`Cancel`, { ns: namespace });
  lang.EDIT_CARE_TEAM = i18n.t(`Edit Care Team`, { ns: namespace });
  lang.CREATE_CARE_TEAM = i18n.t(`Create Care Team`, { ns: namespace });
  lang.EDIT = i18n.t(`Edit`, { ns: namespace });
  lang.CREATE = i18n.t(`Create`, { ns: namespace });
  lang.IDENTIFIER = i18n.t(`Identifier`, { ns: namespace });
  lang.TEAMS = i18n.t(`Care Teams`, { ns: namespace });
  lang.CARE_TEAM_MEMBERS = i18n.t(`Care Teams`, { ns: namespace });
  lang.SEARCH = i18n.t(`Search`, { ns: namespace });
  lang.IDENTIFIER = i18n.t(`Identifier`, { ns: namespace });
  lang.CARE_TEAM_PAGE_HEADER = i18n.t(`FHIR Care Team`, { ns: namespace });
  lang.SUBJECT = i18n.t(`Subject`, { ns: namespace });
  lang.PARTICIPANTS = i18n.t(`Participant`, { ns: namespace });
}

// run it initial
fill();

// bind some events and fill values again (doing the magic you expect to happen magically)
i18n.on(`languageChanged`, () => {
  fill();
});

// export the const
export default lang;
