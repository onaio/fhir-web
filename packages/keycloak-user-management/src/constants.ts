// Web URLs
export const URL_ADMIN = '/admin';
export const URL_USER_EDIT = URL_ADMIN + '/users/edit';
export const URL_USER = URL_ADMIN + '/users/list';
export const URL_USER_GROUPS = `${URL_ADMIN}/users/groups`;
export const URL_USER_GROUP_EDIT = `${URL_ADMIN}/users/groups/edit`;
export const URL_USER_CREATE = URL_ADMIN + '/users/new';
export const URL_USER_GROUP_CREATE = `${URL_ADMIN}/users/groups/new`;
export const URL_USER_CREDENTIALS = URL_ADMIN + '/users/credentials';

// Route params
export const ROUTE_PARAM_USER_ID = 'userId';
export const ROUTE_PARAM_USER_GROUP_ID = 'userGroupId';

// Keycloak API URLs
export const KEYCLOAK_URL_USERS = '/users';
export const KEYCLOAK_URL_USER_GROUPS = '/groups';
export const KEYCLOAK_URL_RESET_PASSWORD = '/reset-password';
export const KEYCLOAK_URL_REQUIRED_USER_ACTIONS = '/authentication/required-actions/';

// OpenSRP API strings
export const OPENSRP_CREATE_PRACTITIONER_ENDPOINT = 'practitioner/user';

// Query patams
export const SEARCH_QUERY_PARAM = 'searchQuery';
