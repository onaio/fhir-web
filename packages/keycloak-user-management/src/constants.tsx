// Web URLs
export const URL_USER_EDIT = '/user/edit';
export const URL_ADMIN = '/admin';
export const URL_USER_CREATE = '/user/new';
export const URL_USER_CREDENTIALS = '/user/credentials';

// Route params
export const ROUTE_PARAM_USER_ID = 'userId';

// Keycloak API URLs
export const KEYCLOAK_URL_USERS = '/users';
export const KEYCLOAK_URL_RESET_PASSWORD = '/reset-password';
export const KEYCLOAK_URL_REQUIRED_USER_ACTIONS = '/authentication/required-actions/';

// Errors
export const ERROR_OCCURED = 'An error occurred';

// Messages
export const USER_DELETED_SUCCESSFULLY = 'User deleted successfully';
