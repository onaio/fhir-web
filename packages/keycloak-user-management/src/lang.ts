export const USER_MANAGEMENT_PAGE_HEADER = 'User Management';
export const ADD_USER = 'Add User';
