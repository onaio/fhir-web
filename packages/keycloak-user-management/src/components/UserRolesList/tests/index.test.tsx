/* eslint-disable @typescript-eslint/camelcase */
import React from 'react';
import reducerRegistry from '@onaio/redux-reducer-registry';
import fetch from 'jest-fetch-mock';
import { authenticateUser } from '@onaio/session-reducer';
import { mount, shallow } from 'enzyme';
import { UserRolesList } from '..';
import { Router } from 'react-router';
import { createBrowserHistory } from 'history';
import toJson from 'enzyme-to-json';
import flushPromises from 'flush-promises';
import { act } from 'react-dom/test-utils';
import * as opensrpStore from '@opensrp/store';
import { Provider } from 'react-redux';
import * as notifications from '@opensrp/notifications';
import {
  reducerName as keycloakUserRolesReducerName,
  reducer as keycloakUserRolesReducer,
  removeKeycloakUserRoles,
} from '../../../ducks/userRoles';
import { userRoles } from '../../../ducks/tests/fixtures';
import { URL_USER_ROLES } from '../../../constants';
import { ERROR_OCCURED } from '../../../lang';
import { unorderedUserRoles } from './fixtures';

jest.mock('@opensrp/store', () => ({
  __esModule: true,
  ...jest.requireActual('@opensrp/store'),
}));

jest.mock('@opensrp/notifications', () => ({
  __esModule: true,
  ...Object.assign({}, jest.requireActual('@opensrp/notifications')),
}));

const history = createBrowserHistory();

const locationProps = {
  history,
  location: {
    hash: '',
    pathname: `${URL_USER_ROLES}`,
    search: '',
    state: {},
  },
  match: {
    isExact: true,
    params: {},
    path: `${URL_USER_ROLES}`,
    url: `${URL_USER_ROLES}`,
  },
};

reducerRegistry.register(keycloakUserRolesReducerName, keycloakUserRolesReducer);

describe('components/UserRolesList', () => {
  beforeEach(() => {
    fetch.resetMocks();
    opensrpStore.store.dispatch(removeKeycloakUserRoles());
  });

  beforeAll(() => {
    opensrpStore.store.dispatch(
      authenticateUser(
        true,
        {
          email: 'bob@example.com',
          name: 'Bobbie',
          username: 'RobertBaratheon',
        },
        // eslint-disable-next-line @typescript-eslint/camelcase
        { api_token: 'hunter2', oAuth2Data: { access_token: 'simple-token', state: 'abcde' } }
      )
    );
  });

  it('renders users table without crashing', () => {
    shallow(
      <Provider store={opensrpStore.store}>
        <Router history={history}>
          <UserRolesList
            {...locationProps}
            keycloakBaseURL="https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage"
          />
        </Router>
      </Provider>
    );
  });
  it('works correctly with store', async () => {
    fetch.once(JSON.stringify(userRoles));
    const props = {
      ...locationProps,
      keycloakBaseURL:
        'https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage',
    };
    const wrapper = mount(
      <Provider store={opensrpStore.store}>
        <Router history={history}>
          <UserRolesList {...props} />
        </Router>
      </Provider>
    );

    await act(async () => {
      await flushPromises();
      wrapper.update();
    });

    expect(wrapper.find('Table')).toBeTruthy();
    expect(wrapper.text()).toMatchSnapshot('full rendered text');
    wrapper.unmount();
  });

  it('renders user list correctly', async () => {
    fetch.once(JSON.stringify(userRoles));
    const props = {
      ...locationProps,
      keycloakBaseURL:
        'https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage',
    };
    const wrapper = mount(
      <Provider store={opensrpStore.store}>
        <Router history={history}>
          <UserRolesList {...props} />
        </Router>
      </Provider>
    );
    // Loader should be displayed
    expect(toJson(wrapper.find('.ant-spin'))).toBeTruthy();

    await act(async () => {
      await flushPromises();
      wrapper.update();
    });
    // Loader should be hiddern
    expect(toJson(wrapper.find('.ant-spin'))).toBeFalsy();

    const userList = wrapper.find('UserRolesList');
    const headerRow = userList.find('Row').at(0);

    expect(headerRow.find('Col').at(0).text()).toMatchSnapshot('header actions col props');
    expect(headerRow.find('Table').first().text()).toMatchSnapshot('table text');
    expect(userList.find('tbody tr')).toHaveLength(5);
    expect(userList.find('tbody').text()).toMatchSnapshot(
      'full table body has 8 user role entries'
    );
    wrapper.unmount();
  });

  it('handles user role list fetch failure', async () => {
    fetch.mockReject(() => Promise.reject('API is down'));
    const mockNotificationError = jest.spyOn(notifications, 'sendErrorNotification');
    const props = {
      ...locationProps,
      keycloakBaseURL:
        'https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage',
    };
    const wrapper = mount(
      <Provider store={opensrpStore.store}>
        <Router history={history}>
          <UserRolesList {...props} />
        </Router>
      </Provider>
    );

    // Loader should be displayed
    expect(toJson(wrapper.find('.ant-spin'))).toBeTruthy();

    await act(async () => {
      await flushPromises();
      wrapper.update();
    });
    expect(mockNotificationError).toHaveBeenCalledWith(ERROR_OCCURED);
  });

  it('shows table with no data if user roles list from api is empty', async () => {
    fetch.once(JSON.stringify([]));
    const props = {
      ...locationProps,
      keycloakBaseURL:
        'https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage',
    };
    const wrapper = mount(
      <Provider store={opensrpStore.store}>
        <Router history={history}>
          <UserRolesList {...props} />
        </Router>
      </Provider>
    );
    // Loader should be displayed
    expect(toJson(wrapper.find('.ant-spin'))).toBeTruthy();

    //Table should be empty
    await act(async () => {
      await flushPromises();
      wrapper.update();
    });

    const userList = wrapper.find('UserRolesList');
    expect(userList.find('Table').first().text()).toEqual('NameCompositeDescriptionNo Data');
  });

  it('sorts by role name', async () => {
    fetch.once(JSON.stringify(unorderedUserRoles));

    const props = {
      ...locationProps,
      keycloakBaseURL:
        'https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage',
    };
    const wrapper = mount(
      <Provider store={opensrpStore.store}>
        <Router history={history}>
          <UserRolesList {...props} />
        </Router>
      </Provider>
    );

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
    });

    wrapper.update();

    // Default order
    expect(wrapper.find('tbody').find('tr').at(0).find('td').at(0).text()).toEqual('ALL_EVENTS');
    expect(wrapper.find('tbody').find('tr').at(1).find('td').at(0).text()).toEqual(
      'PLANS_FOR_USER'
    );
    expect(wrapper.find('tbody').find('tr').at(2).find('td').at(0).text()).toEqual(
      'EDIT_KEYCLOAK_USERS'
    );

    // Ascending
    wrapper.find('thead th').at(0).simulate('click');
    wrapper.update();
    expect(wrapper.find('tbody').find('tr').at(0).find('td').at(0).text()).toEqual('ALL_EVENTS');
    expect(wrapper.find('tbody').find('tr').at(1).find('td').at(0).text()).toEqual(
      'EDIT_KEYCLOAK_USERS'
    );
    expect(wrapper.find('tbody').find('tr').at(2).find('td').at(0).text()).toEqual(
      'PLANS_FOR_USER'
    );

    //Descending
    wrapper.find('thead th').at(0).simulate('click');
    wrapper.update();
    expect(wrapper.find('tbody').find('tr').at(0).find('td').at(0).text()).toEqual(
      'PLANS_FOR_USER'
    );
    expect(wrapper.find('tbody').find('tr').at(1).find('td').at(0).text()).toEqual(
      'EDIT_KEYCLOAK_USERS'
    );
    expect(wrapper.find('tbody').find('tr').at(2).find('td').at(0).text()).toEqual('ALL_EVENTS');
    wrapper.unmount();
  });

  it('sorts by role description', async () => {
    fetch.once(JSON.stringify(unorderedUserRoles));

    const props = {
      ...locationProps,
      keycloakBaseURL:
        'https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage',
    };
    const wrapper = mount(
      <Provider store={opensrpStore.store}>
        <Router history={history}>
          <UserRolesList {...props} />
        </Router>
      </Provider>
    );

    await act(async () => {
      await new Promise((resolve) => setImmediate(resolve));
    });

    wrapper.update();

    // Default order
    expect(wrapper.find('tbody').find('tr').at(0).find('td').at(2).text()).toEqual(
      'Allows on to Download all Events'
    );
    expect(wrapper.find('tbody').find('tr').at(1).find('td').at(2).text()).toEqual(
      'Allows on to view plans for user'
    );
    expect(wrapper.find('tbody').find('tr').at(2).find('td').at(2).text()).toEqual(
      'The role allows one to edit other keycloak users'
    );

    // Ascending
    wrapper.find('thead th').at(2).simulate('click');
    wrapper.update();
    expect(wrapper.find('tbody').find('tr').at(0).find('td').at(2).text()).toEqual(
      'Allows on to Download all Events'
    );
    expect(wrapper.find('tbody').find('tr').at(1).find('td').at(2).text()).toEqual(
      'Allows on to view plans for user'
    );
    expect(wrapper.find('tbody').find('tr').at(2).find('td').at(2).text()).toEqual(
      'The role allows one to edit other keycloak users'
    );

    //Descending
    wrapper.find('thead th').at(2).simulate('click');
    wrapper.update();
    expect(wrapper.find('tbody').find('tr').at(0).find('td').at(2).text()).toEqual(
      'The role allows one to edit other keycloak users'
    );
    expect(wrapper.find('tbody').find('tr').at(1).find('td').at(2).text()).toEqual(
      'Allows on to view plans for user'
    );
    expect(wrapper.find('tbody').find('tr').at(2).find('td').at(2).text()).toEqual(
      'Allows on to Download all Events'
    );
    wrapper.unmount();
  });
});
