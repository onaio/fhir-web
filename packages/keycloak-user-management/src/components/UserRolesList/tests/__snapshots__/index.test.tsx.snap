// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/UserRolesList renders user list correctly: full table body has 8 user role entries 1`] = `"ALL_EVENTSfalseAllows on to Download all EventsEDIT_KEYCLOAK_USERStrueThe role allows one to edit other keycloak usersoffline_accessfalse\${role_offline-access}OPENMRSfalseBasic Role for users, To be changed to User Or ProviderPLANS_FOR_USERfalseAllows on to view plans for user"`;

exports[`components/UserRolesList renders user list correctly: header actions col props 1`] = `"NameCompositeDescriptionALL_EVENTSfalseAllows on to Download all EventsEDIT_KEYCLOAK_USERStrueThe role allows one to edit other keycloak usersoffline_accessfalse\${role_offline-access}OPENMRSfalseBasic Role for users, To be changed to User Or ProviderPLANS_FOR_USERfalseAllows on to view plans for user125 / pageGo to"`;

exports[`components/UserRolesList renders user list correctly: table text 1`] = `"NameCompositeDescriptionALL_EVENTSfalseAllows on to Download all EventsEDIT_KEYCLOAK_USERStrueThe role allows one to edit other keycloak usersoffline_accessfalse\${role_offline-access}OPENMRSfalseBasic Role for users, To be changed to User Or ProviderPLANS_FOR_USERfalseAllows on to view plans for user125 / pageGo to"`;

exports[`components/UserRolesList works correctly with store: full rendered text 1`] = `"User RolesNameCompositeDescriptionALL_EVENTSfalseAllows on to Download all EventsEDIT_KEYCLOAK_USERStrueThe role allows one to edit other keycloak usersoffline_accessfalse\${role_offline-access}OPENMRSfalseBasic Role for users, To be changed to User Or ProviderPLANS_FOR_USERfalseAllows on to view plans for user125 / pageGo to"`;
